libc/src/time/ctime.cpp

```diff
@@ -0,0 +1,26 @@
+//===-- Implementation of ctime function ----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ctime.h"
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+#include "time_utils.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+using LIBC_NAMESPACE::time_utils::TimeConstants;
+
+LLVM_LIBC_FUNCTION(char *, ctime, (const time_t *t_ptr)) {
+  if (t_ptr > TimeConstants::MAXIMUM_32_BIT_VALUE) {
```

Comment:
this is comparing the pointer and not the value, did you mean to do that?

Reply:
Yes, that is correct. The pointer should be dereferenced. Fixed it now.